src/jubjub/mod.rs

```diff
@@ -434,6 +446,257 @@ impl JubjubBls12 {
     }
 }
 
+fn must_write_exp_table(
```

Comment:
Can you add some doc comments to these methods, so it is clear what they are expected to do